### PR TITLE
Preliminary changes for the GN build

### PR DESCRIPTION
**Pull request by Kangz**
These are a bunch of changes that help make BUILD.gn files for Dawn. PTAL!

Most notably it removes the need to generate a proctable per backend, and instead generates one against the frontend.

### CODE REVIEW

File: src/dawn_native/d3d12/DeviceD3D12.cpp [hunk 1]
```diff
@@ -39,14 +39,15 @@
 #include "dawn_native/d3d12/SwapChainD3D12.h"
 #include "dawn_native/d3d12/TextureD3D12.h"
 
-namespace dawn_native { namespace d3d12 {
+namespace dawn_native {
+    dawnProcTable GetProcs();
```
Comment (SenorBlanco) [inline src/dawn_native/d3d12/DeviceD3D12.cpp]: Nit: forward declaring seems a little bit fragile. Is there a reason we can't #include? The header doesn't even have to be autogen'ed, I think.

Comment (Kangz) [reply src/dawn_native/d3d12/DeviceD3D12.cpp]: Agreed, how about I do this in a follow-up patch?

Comment (SenorBlanco) [reply src/dawn_native/d3d12/DeviceD3D12.cpp]: Fine by me. (If I thought it was important, I wouldn't have approved. :) )

**Comment by Kangz** [comment]
Thanks!

**Pull request merged**